src/shared/cli/input/ConsoleInput.php

```diff
@@ -6,9 +6,12 @@ class ConsoleInput implements Input {
     /** @var Output */
     private $output;
 
-    /** @var resource */
+    /** @var false|resource */
```

Comment:
I do understand why this change probably makes psalm happy, but this property can never actually be false if the constructor succeeded...

Maybe we should change the flow to check for false before the assignment?

Reply:
I checked the implementation here, I do agree with you that this should be a resource, but the default value could also be false. So this type hint is correct. Please notice that I added a new check in `\PharIo\Phive\Cli\ConsoleInput::confirm` to check if the `inputStream` is a valid resource. 
Technically `STDIN` can be a `resource` or `false` that's what you assign, I could move the check to the constructor?

Reply:
Let's do that.